compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1928,10 +1929,91 @@ OMR::ARM64::TreeEvaluator::BNDCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg
    return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
    }
 
+static TR::Instruction *compareIntsAndBranchForBNDCHK(TR::ARM64ConditionCode  branchType,
+                                           TR::Node             *node,
+                                           TR::CodeGenerator    *cg,
+                                           TR::SymbolReference  *sr)
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+   TR::LabelSymbol *snippetLabel = generateLabelSymbol(cg);
+   TR::Register *src1Reg = cg->evaluate(firstChild);
+
+   bool foundConst = false;
+   if (secondChild->getOpCode().isLoadConst())
+      {
+      int64_t value = secondChild->get64bitIntegralValue();
+      if (constantIsUnsignedImm12(value))
+         {
+         generateCompareImmInstruction(cg, node, src1Reg, value, true);
+         foundConst = true;
+         }
+      else if (constantIsUnsignedImm12(-value))
+         {
+         generateCompareImmInstruction(cg, node, src1Reg, value, true);
+         foundConst = true;
+         }
+      }
+   if(!foundConst)
+      {
+      TR::Register *src2Reg = cg->evaluate(secondChild);
+      generateCompareInstruction(cg, node, src1Reg, src2Reg, true);
+      }
+
+   TR_ASSERT(!sr, "sr is not expected to be null");
+   cg->addSnippet(new (cg->trHeapMemory()) TR::ARM64HelperCallSnippet(cg, node, snippetLabel, sr));
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, snippetLabel, branchType);
+
+   cg->machine()->setLinkRegisterKilled(true);
+   cg->decReferenceCount(firstChild);
+   cg->decReferenceCount(secondChild);
+   return NULL;
```

Comment:
Change the return value to a non-NULL instruction.  `ArrayCopyBNDCHKEvaluator()` uses the value returned from this function.